tests/test_base_handler.py

```
@@ -0,0 +1,149 @@
+        self.name = self.__class__.__name__
+    
+    async def pre_handle(self, event: BaseEvent) -> BaseEvent:
+        """Pre-process the event before handling."""
+        return event
+    
+    @abstractmethod
+    async def handle(self, event: BaseEvent) -> dict:
+        """Handle the event. Must be implemented by subclasses."""
+        pass
+    
+    async def post_handle(self, event: BaseEvent, result: dict) -> dict:
```

Ruff (0.15.2)

[warning] Unused method argument: `event`

(ARG002)